app/src/main/java/eu/laramartin/medsreminder/common/CalendarUtility.java

```diff
@@ -58,14 +55,45 @@ public static long millisToNextTargetDay(int targetDayOfWeek, int targetHour, in
         return diff;
     }
 
+    @NonNull
+    public static Calendar getNextCalendarDay(int targetDayOfWeek, int targetHour, int targetMinute) {
+        Calendar now = Calendar.getInstance();
+        Calendar next = Calendar.getInstance();
+        int nowDoW = now.get(Calendar.DAY_OF_WEEK);
+        int addDays = (targetDayOfWeek - nowDoW) % 7;
+        next.add(Calendar.DAY_OF_YEAR, addDays);
+        next.set(Calendar.HOUR_OF_DAY, targetHour);
+        next.set(Calendar.MINUTE, targetMinute);
+        return next;
+    }
+
     public static List<Integer> getMedDays(Med med) {
         String medDays = med.getDays();
         return getDaysOfWeek(medDays);
     }
 
-    public static String getFormattedDateWithHour() {
+    public static String getFormattedCurrentDateWithHour() {
```

Comment:
could be simplified by just calling: `return getFormattedDateWithHour(new Date())`